test/integration/dra/binding_conditions_test.go

```diff
@@ -606,3 +557,52 @@ profiles:
 
 	waitForPodScheduled(tCtx, namespace, pod.Name)
 }
+var claimDeviceStatusRetry = wait.Backoff{
+	Steps:    10,
+	Duration: 50 * time.Millisecond,
+	Factor:   1.5,
+	Jitter:   0.1,
+}
+
+// addAllocatedDeviceCondition updates status.devices using the currently allocated device.
+// It retries longer than the default backoff because the scheduler may be updating the same
+// ResourceClaim concurrently when these tests toggle binding conditions.
+func addAllocatedDeviceCondition(tCtx ktesting.TContext, namespace, claimName string, condition metav1.Condition) {
+	tCtx.Helper()
+
+	err := retry.RetryOnConflict(claimDeviceStatusRetry, func() error {
+		latest, err := tCtx.Client().ResourceV1().ResourceClaims(namespace).Get(tCtx, claimName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if latest.Status.Allocation == nil || len(latest.Status.Allocation.Devices.Results) == 0 {
+			return fmt.Errorf("claim %s has no allocated devices to update status for", claimName)
+		}
```

Comment:
> (1) Ensures allocation exists before updating
> ...
> This prevents writing an empty {} device, which caused the 422 error.

But then it doesn't retry. Returning this error causes `retry.RetryOnConflict` to stop, which makes the test fail with this error. Did you expect it to retry?

If this is the root cause of some flakes, then we need to add an explicit `gomega.Eventually` which waits for allocation. I don't think that belongs into `addAllocatedDeviceCondition`.